pvlib/tests/test_tools.py

```diff
@@ -45,6 +45,14 @@ def test__golden_sect_DataFrame_vector():
     v, x = tools._golden_sect_DataFrame(params, lower, upper,
                                         _obj_test_golden_sect)
     assert np.allclose(x, expected, atol=1e-8)
+    # upper and lower bounds equal
+    params = {'c': np.array([1., 2., 1.]), 'n': np.array([1., 1., 1.])}
+    lower = np.array([0., 0.001, 1.])
+    upper = np.array([1., 1.2, 1.])
```

Comment:
I think to replicate the issue reported in #1603, we need `np.all(lower == upper)` here.  Maybe keep this new test case and add another where all the inputs are arrays of length 1?